### ISSUE TITLE
Example cert and key

### ISSUE THREAD

**Issue by hactar** [thpham/portus-registry-tls-compose#1]
Thanks for posting this guide, I'm currently trying to get port.us to run behind a letsencrypt proxy and this guide is proving very useful, especially since you nicely highlight the differences between registry.example.com and hub.example.com, which the official documentation fails to do. One issue that would appear to be a security issue, or maybe I'm misunderstanding something here:

It appears that you are using the example cert  and key from portus:

```ADD https://raw.githubusercontent.com/SUSE/Portus/master/examples/development/vagrant/conf/ca_bundle/server.key /certificates/portus.key```

Isn't that bad? Should one not use a real cert? And if yes, which one is it? Is it the one for hub.example.com or registry.example.com?

**Comment by thpham** [comment]
Hello @hactar, 

thank you for your feedback. Actually you're totally right, it's bad and you should always generate your own certificates !

As far as I understood here, the key is used to generate the JWT token from Portus that the registry will validate with the certificate.

So the flow is:

1. You login to the registry (secured by the letsencrypt SSL with nginx reverse-proxy) by sending your password in base64.
2. The registry forward the request and delegate the authentication to Portus within the internal docker-compose network.
3. Portus check the credentials and then send a response with an encrypted JWT token in the HTTP header.
4. the registry check the signature of the token with the certificate.
5. If the JWT is valid, then (not very sure here) the registry check the authorizations to fetch or not some 
 namespaces (see the notion of team/namespace in Portus)

So basically, I thought that at any moment it's possible to forge a JWT to get access to the registry directly, because the docker client will authenticate with the registry with your password.
And as there is no way to have kind of man-in-the-middle attack within the docker-compose network to inject a forged JWT, I thought it was secure enough.

But I'm not sure when you fetch an image, your docker client split the process in multiple HTTP requests and if the subsequent requests after the authentication are using the JWT.

I will check further and keep you posted, I will do PR as soon as I have more time to explain how to generate our own self-signed certificates.

**Comment by thpham** [comment]
One more note: the token used to fetch the images data maybe also different if you use S3 as storage backend for the registry, then I think you'll get a temporary token from S3 directly.
